libinitializer/SecureInitializer.cpp

```diff
@@ -208,3 +209,343 @@ std::shared_ptr<bas::context> SecureInitializer::SSLContext(Usage _usage)
+        EC_KEY_get0_private_key(ecKey.get()), [](const BIGNUM*) {});
+
+    std::shared_ptr<char> privateKeyData(
+        BN_bn2hex(ecPrivateKey.get()), [](char* p) { OPENSSL_free(p); });
+
+    std::string keyHex(privateKeyData.get());
+    if (keyHex.size() != 64u)
+    {
+        throw std::invalid_argument("Private Key file error! Missing bytes!");
+    }
+
+    KeyPair keyPair = KeyPair(Secret(keyHex));
+
+    std::shared_ptr<boost::asio::ssl::context> sslContext =
+        std::make_shared<boost::asio::ssl::context>(boost::asio::ssl::context::tlsv12);
+
+    std::shared_ptr<EC_KEY> ecdh(
+        EC_KEY_new_by_curve_name(NID_secp256k1), [](EC_KEY* p) { EC_KEY_free(p); });
+    SSL_CTX_set_tmp_ecdh(sslContext->native_handle(), ecdh.get());
+
+    sslContext->set_verify_mode(boost::asio::ssl::context_base::verify_none);
+    INITIALIZER_LOG(INFO) << LOG_BADGE("SecureInitializer") << LOG_DESC("get pub of node")
+                          << LOG_KV("nodeID", keyPair.pub().hex());
+
+    boost::asio::const_buffer keyBuffer(keyContent.data(), keyContent.size());
+    sslContext->use_private_key(keyBuffer, boost::asio::ssl::context::file_format::pem);
+
+    if (!cert.empty() && !contents(cert).empty())
+    {
+        INITIALIZER_LOG(DEBUG) << LOG_BADGE("SecureInitializer") << LOG_DESC("use user certificate")
+                               << LOG_KV("file", cert);
+        sslContext->use_certificate_chain_file(cert);
+    }
+    else
+    {
+        INITIALIZER_LOG(ERROR) << LOG_BADGE("SecureInitializer")
+                               << LOG_DESC("certificate doesn't exist!");
+        BOOST_THROW_EXCEPTION(CertificateNotExists());
+    }
+
+    auto caCertContent = contents(caCert);
+    if (!caCert.empty() && !caCertContent.empty())
+    {
+        INITIALIZER_LOG(DEBUG) << LOG_BADGE("SecureInitializer") << LOG_DESC("use ca certificate")
+                               << LOG_KV("file", caCert);
+
+        sslContext->add_certificate_authority(
+            boost::asio::const_buffer(caCertContent.data(), caCertContent.size()));
+    }
+    else
+    {
+        INITIALIZER_LOG(ERROR) << LOG_BADGE("SecureInitializer")
+                               << LOG_DESC("CA Certificate doesn't exist!");
+        BOOST_THROW_EXCEPTION(CertificateNotExists());
+    }
+
+    if (!caPath.empty())
+    {
+        INITIALIZER_LOG(DEBUG) << LOG_BADGE("SecureInitializer") << LOG_DESC("use ca")
+                               << LOG_KV("file", caPath);
+        sslContext->add_verify_path(caPath);
+    }
+    sslContext->set_verify_mode(boost::asio::ssl::context_base::verify_peer |
+                                boost::asio::ssl::verify_fail_if_no_peer_cert);
+    return ConfigResult{keyPair, sslContext};
+}
+
+ConfigResult initGmConfig()
+{
+    std::string dataPath = "conf/";
+    std::string key = dataPath + "gmnode.key";
+    std::string enKey = dataPath + "gmennode.key";
+    std::string enCert = dataPath + "gmennode.crt";
```

Comment:
read from config file maybe better, will modify this in future

Reply:
thanks